providers/apache/hdfs/pyproject.toml

```
@@ -60,7 +61,8 @@ dependencies = [
     "apache-airflow>=2.10.0",
     'hdfs[avro,dataframe,kerberos]>=2.5.4;python_version<"3.12"',
     'hdfs[avro,dataframe,kerberos]>=2.7.3;python_version>="3.12"',
-    "pandas>=2.1.2,<2.2",
+    'pandas>=2.1.2; python_version <"3.13"',
+    'pandas>=2.2.0; python_version >="3.13"',
```

Just checked release notes... Pandas team claims that 2.2.3 is the first version supporting Python 3.13. See https://pandas.pydata.org/docs/whatsnew/v2.2.3.html#pandas-2-2-3-is-now-compatible-with-python-3-13

Interestingly the 3.13 compatability tag was just added for 2.3.0 in Pypi.

Note I also attempted to `uv pip install pandas==2.2.0` in a Python 3.13 venv and this fails (on x86 as least).
```suggestion
    'pandas>=2.3.0; python_version >="3.13"',
```

---

Do we actually need to add this Python 3.13 specifc pinning here or will the pandas meta data not already select 2.3.0 when installing with Python 3.13?

---

2.2.3 is fine - but I would leave it lower for lower version. We cannot entirely rely on metadata because when go to "lowest" version - it coul be that that version did not yet have != 3.13 exclusion. Most libraries do not add exclusions in advance - so something relased a year ago will not have !=3.13 and when we try to get the lowest version it might fail when building.

---

I will just bump min version to 2.3.3 for 3.13. That should be fine.